microsoft/PowerToys
# [PowerToys Run] Use global HotKey instead of low level keyboard hook

## Summary of the Pull Request

**What is this about:**
Using low level keyboard hooks caused focus issues when invoking PowerToys Run. Using a global HotKey instead solves this issue.

**What is include in the PR:** 
- PowerToys Run handles its own hotkey instead of using the centralized low level keyboard hook.
- Use global shortcut, keeping low level keyboard hook as a fallback.
- Refactor ProcessWindowMessage handler out of EnvironmentHelper, since it's now used for hotkey messages as well.
- Registering HotKey must now be done after there's a valid window handle.
- Properly remove the ProcessWindowMessage handler when the window is closed.

**How does someone test / validate:** 
Verify that PowerToys Run gets focus after being invoked through its HotKey.
This needs to be tested out of the debugger, since focus rules are different when the debugger is involved: https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setforegroundwindow#remarks

## Quality Checklist

- [x] **Linked issue:** #8876 #11817

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
